From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: fgei <fgei@gmail.com>
Date: Fri, 5 Jan 2024 20:28:01 +0000
Subject: [PATCH] Use the old naming convention of "Ads" in ads site settings

The upstream's word choice on blocking ads does not apply in Vanadium.
All sites are blocked with ads unless added as exception on a website
settings.
---
 .../site_settings/ContentSettingsResources.java           | 8 ++++----
 .../browser_ui/site_settings/SingleWebsiteSettings.java   | 6 ++----
 .../components/browser_ui/site_settings/SiteSettings.java | 2 --
 components/browser_ui/strings/android/site_settings.grdp  | 3 +++
 components/page_info/page_info_ui.cc                      | 4 ++--
 components/site_settings_strings.grdp                     | 6 ++++++
 6 files changed, 17 insertions(+), 12 deletions(-)

diff --git a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/ContentSettingsResources.java b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/ContentSettingsResources.java
index f8f4157b5b5c2..316d046f1d626 100644
--- a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/ContentSettingsResources.java
+++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/ContentSettingsResources.java
@@ -107,12 +107,12 @@ public class ContentSettingsResources {
             case ContentSettingsType.ADS:
                 return new ResourceItem(
                         R.drawable.web_asset,
-                        R.string.site_settings_page_intrusive_ads_label,
+                        R.string.ads_permission_title,
                         ContentSettingValues.ALLOW,
                         ContentSettingValues.BLOCK,
-                        R.string.site_settings_page_intrusive_allowed_sub_label,
-                        R.string.site_settings_page_intrusive_blocked_sub_label,
-                        R.string.site_settings_page_intrusive_ads_a11y);
+                        0,
+                        0,
+                        0);
 
             case ContentSettingsType.ANTI_ABUSE:
                 return new ResourceItem(
diff --git a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java
index eee0192f3011e..005c3aa9b7385 100644
--- a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java
+++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java
@@ -1098,10 +1098,8 @@ public class SingleWebsiteSettings extends BaseSiteSettingsFragment
                         && findPreference(assumeNonNull(getPreferenceKey(ContentSettingsType.ADS)))
                                 != null;
 
-        if (!adBlockingActivated) {
-            removePreferenceSafely(PREF_INTRUSIVE_ADS_INFO);
-            removePreferenceSafely(PREF_INTRUSIVE_ADS_INFO_DIVIDER);
-        }
+        removePreferenceSafely(PREF_INTRUSIVE_ADS_INFO);
+        removePreferenceSafely(PREF_INTRUSIVE_ADS_INFO_DIVIDER);
     }
 
     @RequiresNonNull({"mSite"})
diff --git a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SiteSettings.java b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SiteSettings.java
index 9c17b3456b460..822c8b377bc93 100644
--- a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SiteSettings.java
+++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SiteSettings.java
@@ -178,8 +178,6 @@ public class SiteSettings extends BaseSiteSettingsFragment
                 p.setSummary(ContentSettingsResources.getGeolocationAllowedSummary());
             } else if (Type.CLIPBOARD == prefCategory && !checked) {
                 p.setSummary(ContentSettingsResources.getClipboardBlockedListSummary());
-            } else if (Type.ADS == prefCategory && !checked) {
-                p.setSummary(ContentSettingsResources.getAdsBlockedListSummary());
             } else if (Type.SOUND == prefCategory && !checked) {
                 p.setSummary(ContentSettingsResources.getSoundBlockedListSummary());
             } else if (Type.REQUEST_DESKTOP_SITE == prefCategory) {
diff --git a/components/browser_ui/strings/android/site_settings.grdp b/components/browser_ui/strings/android/site_settings.grdp
index 4234408c8a6e0..f92abff051a3c 100644
--- a/components/browser_ui/strings/android/site_settings.grdp
+++ b/components/browser_ui/strings/android/site_settings.grdp
@@ -16,6 +16,9 @@
   <message name="IDS_ALL_SITES" desc='Title of the "All sites" settings screen that allows the user to see permissions for all websites. [CHAR_LIMIT=32]'>
     All sites
   </message>
+  <message name="IDS_ADS_PERMISSION_TITLE" desc="Title for the ads permission [CHAR_LIMIT=32]">
+    Ads
+  </message>
   <message name="IDS_ANTI_ABUSE_PERMISSION_TITLE" desc="Title for the anti-abuse permission [CHAR_LIMIT=32]">
     Auto-verify
   </message>
diff --git a/components/page_info/page_info_ui.cc b/components/page_info/page_info_ui.cc
index 98dd0c356a370..5e7cfb2b17d43 100644
--- a/components/page_info/page_info_ui.cc
+++ b/components/page_info/page_info_ui.cc
@@ -174,8 +174,8 @@ base::span<const PageInfoUI::PermissionUIInfo> GetContentSettingsUIInfo() {
        IDS_SITE_SETTINGS_TYPE_PROTECTED_MEDIA_ID,
        IDS_SITE_SETTINGS_TYPE_PROTECTED_MEDIA_ID_MID_SENTENCE},
 #endif
-      {ContentSettingsType::ADS, IDS_SITE_SETTINGS_TYPE_ADS,
-       IDS_SITE_SETTINGS_TYPE_ADS_MID_SENTENCE},
+      {ContentSettingsType::ADS, IDS_SITE_SETTINGS_TYPE_ADS_2,
+       IDS_SITE_SETTINGS_TYPE_ADS_MID_SENTENCE_2},
       {ContentSettingsType::SOUND, IDS_SITE_SETTINGS_TYPE_SOUND,
        IDS_SITE_SETTINGS_TYPE_SOUND_MID_SENTENCE},
       {ContentSettingsType::CLIPBOARD_READ_WRITE,
diff --git a/components/site_settings_strings.grdp b/components/site_settings_strings.grdp
index 0a64bb60417a1..8b9cfedcdb00b 100644
--- a/components/site_settings_strings.grdp
+++ b/components/site_settings_strings.grdp
@@ -7,6 +7,12 @@
   <message name="IDS_SITE_SETTINGS_TYPE_ADS_MID_SENTENCE" desc="This string is the name of the intrusive ads setting (visit chrome://settings/content and click 'Additional content settings') when used mid-sentence.">
     intrusive ads
   </message>
+  <message name="IDS_SITE_SETTINGS_TYPE_ADS_2" desc="This string is the name of the intrusive ads setting (visit chrome://settings/content and click 'Additional content settings').">
+    Ads
+  </message>
+  <message name="IDS_SITE_SETTINGS_TYPE_ADS_MID_SENTENCE_2" desc="This string is the name of the intrusive ads setting (visit chrome://settings/content and click 'Additional content settings') when used mid-sentence.">
+    ads
+  </message>
   <message name="IDS_SITE_SETTINGS_TYPE_ANTI_ABUSE" desc="The label used for the anti-abuse site settings controls.">
     Auto-verify
   </message>
